pkg/apis/core/validation/validation.go

```diff
@@ -6000,6 +6001,80 @@ func validateWindowsSecurityContextOptions(windowsOptions *core.WindowsSecurityC
+		containerCount++
+		if c.SecurityContext != nil && c.SecurityContext.WindowsOptions != nil && c.SecurityContext.WindowsOptions.HostProcess != nil {
+			if *c.SecurityContext.WindowsOptions.HostProcess {
+				containersWithHostProcessSetTrue = append(containersWithHostProcessSetTrue, c.Name)
+			} else if !*c.SecurityContext.WindowsOptions.HostProcess {
+				containersWithHostProcessSetFalse = append(containersWithHostProcessSetFalse, c.Name)
+			}
+		} else if podHostProcess != nil && *podHostProcess {
+			containersWhichInheritHostProcessTrue = append(containersWhichInheritHostProcessTrue, c.Name)
+		}
+	}
+
+	hostProcessContainerCount := len(containersWithHostProcessSetTrue) + len(containersWhichInheritHostProcessTrue)
+	if hostProcessContainerCount > 0 {
+		// Fail Pod validation if feature is not enabled instead of dropping fields based on PRR reivew.
+		if !utilfeature.DefaultFeatureGate.Enabled(features.WindowsHostProcessContainers) {
+			errMsg := "Pod contains Windows HostProcess containers but feature gate 'WindowsHostProcessContainers' is not enabled"
+			allErrs = append(allErrs, field.Forbidden(fieldPath, errMsg))
+		}
+
+		// If Pod's WindowsOptions.HostProcess is set to false, no container may set WindowsOptions.HostProcess to true.
+		if podHostProcess != nil && !*podHostProcess && len(containersWithHostProcessSetTrue) > 0 {
+			errMsg := fmt.Sprintf("Pod's WindowsOptions.HostProcess is set to false but the following containers override WindowsOptions.HostProcess to true: %v", containersWithHostProcessSetTrue)
+			allErrs = append(allErrs, field.Forbidden(fieldPath, errMsg))
+		}
+
+		// At present, if a Windows Pods contains any HostProcess containers than all containers must be
+		// HostProcess containers (explicitly set or inherited).
+		if hostProcessContainerCount != containerCount {
+			errMsg := "If Pod contains any HostProcess containers then all containers must be HostProcess containers"
+			allErrs = append(allErrs, field.Forbidden(fieldPath, errMsg))
+		}
+
+		// At present Windows Pods which contain HostProcess containers must also set HostNetwork.
+		if hostNetwork != true {
+			errMsg := "PodSecurityContext.HostNetwork must be set if Pod contains any HostProcess containers"
+			allErrs = append(allErrs, field.Forbidden(fieldPath, errMsg))
+		}
+
+		if !capabilities.Get().AllowPrivileged {
+			errMsg := "HostProcess containers are disallowed by cluster policy"
+			allErrs = append(allErrs, field.Forbidden(fieldPath, errMsg))
+		}
+	}
+
+	// If Pod's WindowsOptions.HostProcess is set to true, no contianer may set WindowsOptions.HostProcess to false.
+	if podHostProcess != nil && *podHostProcess && len(containersWithHostProcessSetFalse) > 0 {
+		errMsg := fmt.Sprintf("Pod's WindowsOptions.HostProcess is set to true but the following containers override WindowsOptions.HostProcess to false: %v", containersWithHostProcessSetFalse)
+		allErrs = append(allErrs, field.Forbidden(fieldPath, errMsg))
+	}
```

Comment:
I think this would be covered by the `if podHostProcess != nil && containerHostProcess != nil && *podHostProcess != *containerHostProcess` check inside the container loop as well